src/parse_execution.cpp

```diff
@@ -749,10 +749,40 @@ end_execution_reason_t parse_execution_context_t::handle_command_not_found(
             event_args.insert(event_args.begin(), cmd_str);
         }
 
-        event_fire_generic(*parser, L"fish_command_not_found", &event_args);
+        wcstring buffer;
+        wcstring error;
+
+        // Redirect to stderr
+        auto io = io_chain_t{};
+        io.append_from_specs({redirection_spec_t{STDOUT_FILENO, redirection_mode_t::fd, L"2"}}, L"/");
```

Comment:
`L"/"` could even be the empty string since it's never used, right? A shame we can't have NULL here :older_man: